databuilder/loader/file_system_neptune_csv_loader.py

```diff
@@ -0,0 +1,179 @@
+        while node:
+
+            node.attributes[PUBLISHED_TAG_PROPERTY_NAME] = self.job_publisher_tag
+            node_dict = neptune_serializer.convert_node(node)
+            if node_dict:
+                key = (node.label, len(node_dict))
+                file_suffix = '{}_{}'.format(*key)
+                node_writer = self._get_writer(
+                    node_dict,
+                    self._node_file_mapping,
+                    key,
+                    self._node_dir,
+                    file_suffix
+                )
+                node_writer.writerow(node_dict)
+            node = csv_serializable.next_node()
+
+        relation = csv_serializable.next_relation()
+        while relation:
+            relation.attributes[PUBLISHED_TAG_PROPERTY_NAME] = self.job_publisher_tag
+            relation_dicts = neptune_serializer.convert_relationship(relation)
+            if relation_dicts:
+                key2 = (relation.start_label,
+                        relation.end_label,
+                        relation.type,
+                        len(relation_dicts[0]))
+
+                file_suffix = '{}_{}_{}'.format(key2[0], key2[1], key2[2])
+                relation_writer = self._get_writer(relation_dicts[0],
+                                                   self._relation_file_mapping,
+                                                   key2,
+                                                   self._relation_dir,
+                                                   file_suffix)
+                relation_writer.writerows(relation_dicts)
+            relation = csv_serializable.next_relation()
+
+    def _get_writer(self,
+                    csv_record_dict: Dict[str, Any],
+                    file_mapping: Dict[Any, DictWriter],
+                    key: Any,
+                    dir_path: str,
+                    file_suffix: str
+                    ) -> DictWriter:
+        """
+        Finds a writer based on csv record, key.
+        If writer does not exist, it's creates a csv writer and update the
+        mapping.
+        """
+        writer = file_mapping.get(key)
+        if writer:
+            return writer
+
+        LOGGER.info('Creating file for {}'.format(key))
+
+        file_out = open('{}/{}.csv'.format(dir_path, file_suffix), 'w', encoding='utf8')
+        writer = csv.DictWriter(file_out, fieldnames=csv_record_dict.keys(), quoting=csv.QUOTE_NONNUMERIC)
+
+        def file_out_close() -> None:
+            LOGGER.info('Closing file IO {}'.format(file_out))
+            file_out.close()
+        self._closer.register(file_out_close)
+
+        writer.writeheader()
+        file_mapping[key] = writer
+
+        return writer
+
+    def close(self) -> None:
+        """
+        Any closeable callable registered in _closer, it will close.
+        """
+        self._closer.close()
+
+    def get_scope(self) -> str:
+        return "loader.filesystem_csv_neptune"
```

Comment:
loader.neptune_filesystem_csv ?

Reply:
I was trying to be consistent with https://github.com/amundsen-io/amundsendatabuilder/blob/master/databuilder/loader/file_system_neo4j_csv_loader.py#L187

but I think loader.neptune_filesystem_csv sounds better.